pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -228,8 +255,69 @@ func (r *ReconcileIBMBlockCSI) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileIBMBlockCSI) addFinalizerIfNotPresent(instance *ibmblockcsi.IBMBlockCSI) error {
+	logger := log.WithName("addFinalizer")
+
+	accessor, finalizerName, err := r.getAccessorAndFinalizerName(instance)
+	if err != nil {
+		return err
+	}
+
+	if !util.Contains(accessor.GetFinalizers(), finalizerName) {
+		logger.Info("adding", "finalizer", finalizerName, "on", accessor.GetName())
+		accessor.SetFinalizers(append(accessor.GetFinalizers(), finalizerName))
+
+		if err := r.client.Update(context.TODO(), instance.Unwrap()); err != nil {
+			logger.Error(err, "failed to add", "finalizer", finalizerName, "on", accessor.GetName())
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileIBMBlockCSI) hasFinalizer(instance *ibmblockcsi.IBMBlockCSI) (bool, error) {
+	accessor, finalizerName, err := r.getAccessorAndFinalizerName(instance)
+	if err != nil {
+		return false, err
+	}
+
+	if util.Contains(accessor.GetFinalizers(), finalizerName) {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (r *ReconcileIBMBlockCSI) removeFinalizer(instance *ibmblockcsi.IBMBlockCSI) error {
+	logger := log.WithName("removeFinalizer")
+
+	accessor, finalizerName, err := r.getAccessorAndFinalizerName(instance)
+	if err != nil {
+		return err
+	}
+
+	accessor.SetFinalizers(util.Remove(accessor.GetFinalizers(), finalizerName))
+	if err := r.client.Update(context.TODO(), instance.Unwrap()); err != nil {
```

Comment:
I suggest breaking this line to two lines:
```suggestion
	err := r.client.Update(context.TODO(), instance.Unwrap())
	if err != nil {
```
since this line does too many things:
1. `if` statement
2. local variable declaration
3. client updating

WDYT?

Reply:
it's just a common syntax suger, both look ok to me